src/Raven.Server/Documents/Replication/Incoming/AbstractIncomingReplicationHandler.cs

```diff
@@ -448,6 +463,7 @@ protected void ReceiveSingleDocumentsBatch(TOperationContext context, int replic
                 {
                     using (var networkStats = stats.For(ReplicationOperation.Incoming.Network))
                     {
+                        RecordDatabaseChangeVector(context, networkStats);
```

Comment:
you record it before you get the batch? it is a bit racy and might give us misleading info. 
I think I would prefer to record it under the write lock (not sure if before or after the batch though)

Reply:
I think so, yes

Reply:
done